subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/preconditions/IntegTestPreconditions.groovy

```diff
@@ -60,19 +70,47 @@ class IntegTestPreconditions {
     }
 
     static final class NotDaemonExecutor implements TestPrecondition {
+        @Override
+        boolean isSatisfied() throws Exception {
+            return !GradleContextualExecuter.isDaemon()
+        }
+    }
+
+    static final class IsNoDaemonExecutor implements TestPrecondition {
         @Override
         boolean isSatisfied() throws Exception {
             return GradleContextualExecuter.isNoDaemon()
         }
     }
 
+    static final class NotNoDaemonExecutor implements TestPrecondition {
```

Comment:
Why isn't this equal to `IsDaemonExecutor`? In any case double negatives are hard to understand.